src/tribler-core/tribler_core/modules/libtorrent/stream.py

```diff
@@ -1,182 +1,622 @@
+
+2-DYNMAIC PRIORITISATION: When the static prio is finished then the client can start playing the file,
+When a client starts playing the file over http, it will request a chunk, each requested chunk will initiate a
+dynmaic buffer according to the current read position of the file mapped to related piece of the torrent file.
+The undownloaded pieces starting from the current read position with the length of prebuffsize will be prioritised
+with the DEADLINE_PRIO_MAP sequence and will be deadlined with the indexes of the same map.
+Rest of the pieces are to prio: 1 and no deadline. Note that, the prio, deadline and the actual pieces impacted
+will be dynamically updated eachtime more chunks are readed, until EOF.
+Each chunk will have its own prio applied, and there can be multiple concurrent chucks
+for a given fileindex of a torrent
+"""
+
 import logging
 from asyncio import sleep
 
+from tribler_common.simpledefs import DLSTATUS_DOWNLOADING, DLSTATUS_SEEDING
+
+from tribler_core.modules.libtorrent import check_vod
 from tribler_core.utilities.torrent_utils import get_info_from_handle
 
+# Header and footer sizes are necessary for video client to detect file codecs and muxer metadata.
+# Without below pieces are ready, streamer should not start
+HEADER_SIZE = 5 * 1024 * 1024
+FOOTER_SIZE = 1 * 1024 * 1024
+# the percent of the file size to be used as moving or static prebufferng size
+PREBUFF_PERCENT = 0.05
+# Below map defines the priority/deadline sequence for pieces. List index represents the deadline, and
+# actual value in the index represents the priority of the piece sequence in the torrent file.
+# Minimum prio must be 2, because prio 1 is used for not relevant pieces in streaming
+# Max prio is 7 and must have only 1 deadline attached to it, because lt selects prio 7 regardles of picker desicion
+# The narrower this map, better the prioritaztion worse the throughput ie: 7,6,5,3,2,1
+# The wider this map, worse the prioritization, better the throughput: ie: 7,6,6,6,6,6,6,6,6,5,5,5,5,5,5,5,5,5,5...
+# Below logarithmic map is based on 2^n and achieves 3~4MB/s with a moving 4~5MB window prioritization
+# There is no prio 5 in libtorrent priorities
+# https://www.libtorrent.org/manual.html#piece-priority-prioritize-pieces-piece-priorities
+DEADLINE_PRIO_MAP = [7, 6, 6, 4, 4, 4, 4, 3, 3, 3, 3, 3, 3, 3, 3, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
+# time to detect if the client is no more requesting stream chunks from rest api
+# basically this means the video is either paused, or seeked to another postion
+# but the player still wants the old tcp session alive,
+# above setting can be reduced as low as 1 sec.
+# lower this value, better the seek responsivenes
+STREAM_PAUSE_TIME = 1
+# never use 0 priority because when streams are paused
+# we still want lt to download the pieces not important for the stream
+MIN_PIECE_PRIO = 1
+
+
+class ERROR_NOT_STREAMING(Exception):
```

Comment:
Please stick to the normal way we name classes (`ERROR_NOT_STREAMING` -> `NotStreamingError`).

Reply:
ok